tests/templates/test_subroutines/test_fable.py

```diff
@@ -285,6 +285,29 @@ def circuit_jax(input_matrix):
         gradient_jax = grad_fn(input_matrix_jax)
         assert np.allclose(gradient_numeric, gradient_jax[0, 0], rtol=0.001)
 
+    @pytest.mark.jax
+    def test_jit_result(self):
```

Comment:
How come the gradient test with JIT was passing initially?

Reply:
It seems `test_fable_grad_jax_jit` is testing the gradient computed with JIT against this

``
       gradient_numeric = (
            circuit_jax(input_jax_positive_delta) - circuit_jax(input_jax_negative_delta)
        ) / (2 * delta)
``

but `circuit_jax` is defined with the JIT decorator, so in the assert two values computed with JIT are being compared and the bug is not revealed.

Reply:
Then that test needs to be corrected too. The test currently is:
```
    def test_fable_grad_jax_jit(self, input_matrix):
        """Test that FABLE is differentiable when using jax."""
```
We can compute the reference gradients with a new circuit that is not JITed. Alternatively, we might split the test into two tests: one that computes the gradients with JAX and one that computes the gradients with JIT.

Reply:
I changed the test to compare the numeric gradient computed with a non-JITed function with the JAX gradient of the JITed function. I don't think we need to split into two tests since the test `test_fable_grad_jax` already tests the JAX gradient.